src/training/ea/lipizzaner_gan_trainer.py

```diff
@@ -140,19 +140,23 @@ def train(self, n_iterations, stop_event=None):
             new_populations = {}
 
             # Create random dataset to evaluate fitness in each iterations
-            fitness_samples = self.generate_random_fitness_samples(self.fitness_sample_size)
+            fitness_input, fitness_labels = self.generate_random_fitness_samples(self.fitness_sample_size)
```

Comment:
What have you changed fitness_samples to fitness_input? 
I like more fitness_samples name, isn't it?